stablehlo/reference/InterpreterApi.h

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_REFERENCE_INTERPRETERAPI_H
+#define STABLEHLO_REFERENCE_INTERPRETERAPI_H
+
+#include "llvm/Support/Error.h"
+#include "llvm/Support/ErrorOr.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Support/LLVM.h"
+#include "stablehlo/reference/InterpreterValue.h"
+#include "stablehlo/reference/Process.h"
+#include "stablehlo/reference/Scope.h"
+
+namespace mlir {
+namespace stablehlo {
+
+/// Base interpreter fallback callback functor to run when no registered kernels
+/// are found for a given StableHLO operation.
+struct InterpreterConfiguration;
+struct InterpreterFallback {
+  llvm::Error operator()(Operation &op, Process *process, Scope &scope);
+
+  virtual ~InterpreterFallback() = default;
+
+  /// Set the user provided interpreter configuration.
+  void setConfig(const InterpreterConfiguration &cfg) { config = &cfg; }
+
+  /// Set the topmost currently executing function in the module.
+  void setFcn(func::FuncOp fcn) { currentFcn = fcn; }
```

Comment:
Proposal:
```
void setFunction(func::FuncOp funcOp) { currentFunction = funcOp; }
```

Abbreviating this does not give us much mileage, plus the code in the implementation
```
config.fallback->setFcn(funcOp);
```
would read nicely:
```
config.fallback->setFunction(funcOp);
```